RollbarNotifier/Sources/RollbarReport/Formatter/FormatBuilder.swift

```diff
@@ -159,11 +158,110 @@ extension Formatted {
     }
 }
 
+extension Formatted {
+    /// Concatenates two formatted optional strings.
+    @inlinable
+    static func *? <Success: StringProtocol, Failure: Error>(lhs: Self, rhs: Self) -> Formatted<Result<String, Failure>>
+        where T == Result<Success, Failure> {
+        switch (lhs.string, rhs.string) {
+        case let (.failure(failure), _),
+             let (_, .failure(failure)):
+            Formatted<Result<String, Failure>>(string: .failure(failure))
+        case (.none, _),
+             (_, .none):
+            Formatted<Result<String, Failure>>(string: .none)
+        case let (.success(lhs), .success(rhs)):
+            Formatted<Result<String, Failure>>(string: .success(lhs.appending(rhs)))
+        }
+    }
+}
```

Comment:
So, here's where the magic happens, right? 

If I understand correctly, you're adding the idea that a `T` can be a `Result` in order to return a `Failure` in case of an overflow, is that correct?

Ideally, what we'd want is to return some error string, maybe to avoid all the workaround dance. (Not asking you to do it, or implying what you did is wrong, just thinking out loud to see what you think, too!)